app/uk/gov/hmrc/agentepayeregistration/validators/AgentEpayeRegistrationValidator.scala

```diff
@@ -0,0 +1,98 @@
+package uk.gov.hmrc.agentepayeregistration.validators
+
+import cats.Semigroup
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import uk.gov.hmrc.agentepayeregistration.models.{Failure, RegistrationRequest}
+
+object ValidatedSemigroup {
+  implicit def validatedSemigroup[A] = new Semigroup[Validated[Failure, Unit]] {
+    def combine(x: Validated[Failure, Unit], y: Validated[Failure, Unit]): Validated[Failure, Unit] = (x, y) match {
+      case (Valid(_), Valid(_)) => Valid(())
+      case (Invalid(f1),Invalid(f2)) => Invalid(Failure(f1.errors ++ f2.errors))
+      case (Valid(_), f@Invalid(_)) => f
+      case (f@Invalid(_), Valid(_)) => f
+    }
+  }
+}
+
+object AgentEpayeRegistrationValidator {
+  import ValidatedSemigroup._
+
+  def validate(request: RegistrationRequest): Validated[Failure, Unit] = {
+
+    def mandatoryChars(field: String, propertyName: String, limit: Int) =
+      nonEmpty(field)(propertyName)
+        .andThen(_ => isValidCharacters(field)(propertyName))
+        .andThen(_ => maxLength(field, limit)(propertyName))
+
+    def mandatoryPostcode(field: String, propertyName: String) =
+      nonEmpty(field)("postcode").andThen(_ => isPostcode(field)("postcode"))
+
+    val validators = Seq(
+      mandatoryChars(request.agentName, "agent name", 56),
+      mandatoryChars(request.contactName, "contact name", 56),
+      mandatoryChars(request.address.addressLine1, "address line 1", 35),
+      mandatoryChars(request.address.addressLine2, "address line 2", 35),
+      mandatoryPostcode(request.address.postCode, "postcode")
+    )
+
+    def numberWithLimit(field: String, propertyName: String, limit: Int) =
+      isInteger(field)(propertyName).andThen(_ => maxLength(field, limit)(propertyName))
+
+    def validCharsWithLimit(field: String, propertyName: String, limit: Int) =
+      isValidCharacters(field)(propertyName).andThen(_ => maxLength(field, limit)(propertyName))
+
+    def emailValidatorWithLimit(field: String, propertyName: String, limit: Int) =
+      isEmailAddress(field)(propertyName).andThen(_ => maxLength(field, limit)(propertyName))
+
+    val optionalFieldValidators = Seq(
+      request.telephoneNumber.map(x => numberWithLimit(x, "telephone number", 35)),
```

Comment:
The validation on telephone numbers wont allow these prefix as per the specifications. The validators only allow numbers. (Clarified with Nick)